New features on product-rating-summary

khrizzcristian opened the pull request:
#### What problem is this solving?
I added new features on product-rating-summary. They are all active on the workspace:

1. Ability to display stars even if the product has no reviews
2. Ability to hide the total number of reviews in product-rating-summary block
3. Ability to add `Add Review` button under the stars in product-rating-summary block


<!--- Don't forget to add a link to a Workspace where this branch is linked -->

[Workspace](https://longo--customsoft.myvtex.com/ceas-crystal-frost-ls-hthtsts-5484070/p)

klzns commented on messages/en.json:
```diff
@@ -94,5 +94,9 @@
   "store/reviews.list.anonymous": "Anonymous",
-  "store/reviews.list.pagination.textOf": "of"
+  "store/reviews.list.pagination.textOf": "of",
+  "store/reviews.list.graph.stars": "{value} stars",
```
You can use pluralization feature of ICU format

```
{value, plural, =0{0 stars} =1{1 star} other{# stars}}
```

You can read more here:
https://format-message.github.io/icu-message-format-for-translators/

klzns commented on messages/es.json:
```diff
@@ -94,5 +94,9 @@
   "store/reviews.list.anonymous": "Anónimo",
-  "store/reviews.list.pagination.textOf": "de"
+  "store/reviews.list.pagination.textOf": "de",
+  "store/reviews.list.graph.stars": "Extrellas",
```
Missing variable value

klzns commented on messages/pt.json:
```diff
@@ -94,5 +94,9 @@
   "store/reviews.list.anonymous": "Anônimo",
-  "store/reviews.list.pagination.textOf": "de"
+  "store/reviews.list.pagination.textOf": "de",
+  "store/reviews.list.graph.stars": "Estrelas",
```
Same

klzns commented on react/RatingSummary.tsx:
```diff
@@ -11,7 +11,12 @@ import { NormalizedCacheObject } from 'apollo-cache-inmemory'
 import { useCssHandles } from 'vtex.css-handles'
+import { FormattedMessage } from 'react-intl'
+import { Link, canUseDOM } from 'vtex.render-runtime'
+import path from 'ramda/es/path'
```
Ramda is a global package in IO, you can use this format:

```suggestion
import { path } from 'ramda'
```

klzns commented on react/RatingSummary.tsx:
```diff
@@ -115,13 +181,53 @@ const RatingSummary: FunctionComponent<Props> = props => {
+      }
+
+      const { namespaces } = sessionRespose
+      const storeUserId = path(
+        ['authentication', 'storeUserId', 'value'],
+        namespaces
+      )
```
You can avoid ramda with optional chaining in TS files
```suggestion
      const storeUserId = namespaces?.authentication?.storeUserId?.value
```

klzns commented on react/Reviews.tsx:
```diff
@@ -477,9 +490,18 @@ const Reviews: FunctionComponent<InjectedIntlProps & Props> = props => {
+        graphArray[0] = total
+        if (reviews) {
+          // eslint-disable-next-line array-callback-return
+          reviews.map((review: Review) => {
```
```suggestion
          reviews.forEach((review: Review) => {
```

klzns commented on react/Reviews.tsx:
```diff
@@ -477,9 +490,18 @@ const Reviews: FunctionComponent<InjectedIntlProps & Props> = props => {
+        const graphArray = [0, 0, 0, 0, 0, 0]
+        graphArray[0] = total
+        if (reviews) {
+          // eslint-disable-next-line array-callback-return
```
You can remove this if you use forEach
```suggestion
```

klzns commented on react/Reviews.tsx:
```diff
@@ -533,6 +558,170 @@ const Reviews: FunctionComponent<InjectedIntlProps & Props> = props => {
         )}
       </div>
+      {state.settings.showGraph ? (
+        <div className={`${handles.graphContent} mv5`}>
```
Create a separate component for the graph

khrizzcristian replied:
Done :)
Link active on: https://longo--biasicom.myvtex.com/aspirator-cu-sac-samsung-vc07m25e0wr-2-5-l-700-w-tub-telescopic-accesorii-2-in-1-rosu/p

klzns commented on CHANGELOG.md:
```diff
@@ -5,6 +5,13 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+New features: 
+1. Ability to display stars even if the product has no reviews
+2. Ability to hide the total number of reviews in product-rating-summary block
+3. Ability to add `Add Review` button under the stars in product-rating-summary block
+
+### Added
+- Display graph option in app settings
```
```suggestion
### Added
- Display graph option in app settings
- New settings to display stars even if the product has no reviews
- New settings to hide the total number of reviews in product-rating-summary block
- New settings to add `Add Review` button under the stars in product-rating-summary block
```

guerreirobeatriz commented on docs/README.md:
```diff
@@ -38,6 +38,12 @@ Once in the app's settings page, define the following settings:
 
 - **Number of open review accordions** - Checking this box allows you to set a specific number of review accordions (instead of all of them) to automatically open when the page is loaded, displaying all the review text. If the `Default all review accordions to open` setting is active, this option is ignored.
 
+- **Display graph** - Checking this box allows you to display the reviews graph on the product page.
+
+- **Display stars in product-rating-summary if there are no reviews** - Checking this box allows you to display empty stars if the product has no reviews
+- **Display total reviews number on product-rating-summary block** - Checking this box allows you to display the total number of product reviews.
+- **Display `Add review` button on product-rating-summary block** - Checking this box allows you to display an Add review button under the stars.
```
```suggestion
- **Display graph** - Checking this box allows you to display the reviews graph on the product page.

- **Display stars in `product-rating-summary` if there are no reviews** - Checking this box allows you to display empty stars even if the product still has no reviews.

- **Display total reviews number on `product-rating-summary` block** - Checking this box allows you to display the total number of product reviews.

- **Display `Add review` button on `product-rating-summary` block** - Checking this box allows you to display an `Add review` button under the stars.
```

guerreirobeatriz commented on docs/README.md:
```diff
@@ -87,6 +95,14 @@ In order to apply CSS customizations in this and other blocks, follow the instru
 | `starsContainer`          |
 | `summaryContainer`        |
 | `writeReviewContainer`    |
+| `graphContent`            |
+| `graphContainer`          |
+| `graphText`               |
+| `graphTextLabel`          |
+| `graphBarContainer`       |
+| `graphBar`                |
+| `graphBarPercent`         |
+| `summaryButtonContainer`  |
```
This should be added to the table following the alphabetical order!

guerreirobeatriz replied:
Sorry for not pointing this out in my last review 😢

khrizzcristian commented:
Can you review this pull request pls? We are in stand by right now because we can't implement this

polishq commented:
> Can you review this pull request pls? We are in stand by right now because we can't implement this

There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.

khrizzcristian commented:
> > Can you review this pull request pls? We are in stand by right now because we can't implement this
> 
> There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.

Done :)

polishq commented:
> > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > 
> > 
> > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> 
> Done :)

Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.

khrizzcristian commented:
> > > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > > 
> > > 
> > > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> > 
> > 
> > Done :)
> 
> Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.

Rebased

polishq commented:
> > > > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > > > 
> > > > 
> > > > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> > > 
> > > 
> > > Done :)
> > 
> > 
> > Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.
> 
> Rebased

@khrizzcristian It looks like you performed a merge rather than a rebase. Please redo the rebase using these commands:
git checkout master
git checkout pull
git checkout -
git rebase master
git push --force

I also see that the Danger CI bot is throwing a failure. Not sure what that's about but let's see if the rebase fixes it.

khrizzcristian commented:
> > > > > > Can you review this pull request pls? We are in stand by right now because we can't implement this
> > > > > 
> > > > > 
> > > > > There have been a few releases since this PR was opened. Please rebase your branch and I'll add my approval.
> > > > 
> > > > 
> > > > Done :)
> > > 
> > > 
> > > Please rebase from master. Your `manifest.json` shows version 2.2.1, but current version is 2.3.1.
> > 
> > 
> > Rebased
> 
> @khrizzcristian It looks like you performed a merge rather than a rebase. Please redo the rebase using these commands:
> git checkout master
> git checkout pull
> git checkout -
> git rebase master
> git push --force
> 
> I also see that the Danger CI bot is throwing a failure. Not sure what that's about but let's see if the rebase fixes it.

It shows me that everything is up to date. Rebased, pushed, pulled, etc...
My manifest.json shows "version": "2.3.2"

polishq commented:
> It shows me that everything is up to date. Rebased, pushed, pulled, etc...
> My manifest.json shows "version": "2.3.2"

It's a common mistake, I've done it myself in the past. In a true rebase the commits that are in master are not shown here on this PR. We need that to be resolved.

khrizzcristian commented:
> > It shows me that everything is up to date. Rebased, pushed, pulled, etc...
> > My manifest.json shows "version": "2.3.2"
> 
> It's a common mistake, I've done it myself in the past. In a true rebase the commits that are in master are not shown here on this PR. We need that to be resolved.

Rebased again :)

klzns commented on messages/es.json:
```diff
@@ -94,5 +94,8 @@
   "store/reviews.list.anonymous": "Anónimo",
-  "store/reviews.list.pagination.textOf": "de"
+  "store/reviews.list.pagination.textOf": "de",
+  "store/reviews.list.graph.stars": "{value, plural, =0{0 extrellas} =1{1 extrella} other{# extrellas}}",
```
```suggestion
  "store/reviews.list.graph.stars": "{value, plural, =0{0 estrellas} =1{1 estrella} other{# estrellas}}",
```
